### PR TITLE
[releng] 1.20: Add Slack exempt_branches rule for release-1.20 branch

### PR DESCRIPTION
**Pull request by xmudrii**
* releng: Add Slack exempt_branches rule for release-1.20 branch
* releng: Update Release Managers for Slack exempt_branches

Pending branch cut:
/hold

/sig release
/area release-eng
/assign @saschagrunert @cpanato
cc: @kubernetes/release-engineering

### CODE REVIEW

File: config/prow/plugins.yaml
Comment (cpanato) [inline config/prow/plugins.yaml:249]: I think we need to keep this for now, since 1.17 is not deprecated yet. not 100% sure

Comment (xmudrii) [reply config/prow/plugins.yaml:249]: I've thought the same, but the [branch management handbook](https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/branch-manager.md#update-slack-branch-whitelists) says:
> Remove the oldest release branch block

I'd say that we leave 1.17 and remove it at some point later, but I'm not sure is that okay.

Comment (saschagrunert) [reply config/prow/plugins.yaml:249]: Yep, end of Life for 1.17 is 2021-01-30

Comment (xmudrii) [reply config/prow/plugins.yaml:249]: @cpanato @saschagrunert PTAL, updated.

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by xmudrii** [comment]
I've removed Tim, as Tim is not a release manager anymore.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by xmudrii** [comment]
The branch has been cut!
/hold cancel

**Pull request merged**
